File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxi = INT_MIN;
        int current = 0;
        for(int i=0;i<nums.size();i++){
            current+= nums[i];
            if(current>maxi){
                maxi = current;
            }
            if(current<0){
                current = 0;
            }
        }
        return maxi;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxi = Integer.MIN_VALUE;
        int curr = 0;
        if(nums.length<=1){
            return nums[0];
        }
        for(int i=0;i<nums.length;i++){
            curr+=nums[i];
            if(maxi<curr){
                maxi = curr;
            }
            if(curr<0){
                curr= 0;
            }
        }
        return maxi;
    }
}